pkg/modules/auth/openid/openid.go

```diff
@@ -280,13 +292,46 @@ func getOrCreateUser(s *xorm.Session, cl *claims, provider *Provider, idToken *o
 		if cl.Name != u.Name {
 			u.Name = cl.Name
 		}
+		if cl.Picture != "" && u.AvatarProvider == "" {
```

Comment:
Doesn't this mean if the user has already set their avatar to something else, their avatar from the openid provider won't get synced? Since `u.AvatarProvider` will be empty in that case.

Reply:
This line of code also has no effect, since the provider field will always have a default value and should never be empty.

The goal of my change is to avoid overwriting the avatars of OpenID users who have previously changed them.
Only new users or those who still use the default initials avatar will be affected.

If default avatar provider in config.yml is updated to another provider, all OpenID users in that instance will not be affected by this change.

I to minimize impact on existing user data. Additionally, because of EmailFallback, we cannot guarantee that an OpenID user will always log in via OpenID.

I will push a fix later

Reply:
> The goal of my change is to avoid overwriting the avatars of OpenID users who have previously changed them

Why? Shouldn't the provider always take precedence over what the user wants? That would be in line with not being able to change the avatar later in Vikunja.

> If default avatar provider in config.yml is updated to another provider, all OpenID users in that instance will not be affected by this change.

The default settings are only taken into account during user registration, existing accounts are never affected by a change there.

Reply:
Done, the provider check is removed